### PR TITLE
refactor(2671): Moved retry logic from infer_type_name to wizard

### LINKED ISSUE #2671: refactor: Move `retry` logic from `infer_type_name` to `wizard`
- Move the logic to wizard
- Use tokio_retry instead of our existing implementation to retry with an exponential back off if the the API returns a 429 error.

### PR DESCRIPTION
**Pull request by harshtech123**
/claim #2671

**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #2671 

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

**Comment by tusharmath** [comment]
Unnecessary files have been deleted. Please revert.

File: src/cli/llm/infer_type_name.rs [hunk 1]
```diff
@@ -102,83 +98,22 @@ impl InferTypeName {
...
-                match answer {
-                    Ok(answer) => {
-                        let name = &answer.suggestions.join(", ");
-                        for name in answer.suggestions {
-                            if config.types.contains_key(&name)
-                                || new_name_mappings.contains_key(&name)
-                            {
-                                continue;
-                            }
+            match wizard.ask_with_retry(question).await {
```
Comment (tusharmath) [inline src/cli/llm/infer_type_name.rs:101]:
```suggestion
            match wizard.ask(question).await {
```

File: src/cli/llm/infer_type_name.rs [hunk 2]
```diff
@@ -102,83 +98,22 @@ impl InferTypeName {
...
-        let resp = ChatResponse {
-            content: Some(MessageContent::Text(
-                "{\"suggestions\":[\"Post\",\"Story\",\"Article\",\"Event\",\"Brief\"]}".to_owned(),
-            )),
-            ..Default::default()
-        };
-        let answer = Answer::try_from(resp).unwrap();
-        insta::assert_debug_snapshot!(answer);
-    }
-}
```
Comment (tusharmath) [inline src/cli/llm/infer_type_name.rs:184]: Don't delete existing tests.

Comment (harshtech123) [reply src/cli/llm/infer_type_name.rs:184]: restored

**Review by harshtech123 — COMMENTED** [review]
done with updates

**Review by harshtech123 — COMMENTED** [review]
uodated @tusharmath

**Comment by tusharmath** [comment]
closing in favour of https://github.com/tailcallhq/tailcall/pull/2673

**Comment by tusharmath** [comment]
@harshtech123 I think you are going mostly in the right direction but end up pushing changes with unnecessary file updates. Please feel free to create a new PR that's cleaner.

**Comment by harshtech123** [comment]
@tusharmath i have created a new pr , kindly check that one 
#2680

**Pull request closed without merging**